### ISSUE TITLE
Terraform v0.11.2 with provider.aws v1.7.0 doesn't support region cn-northwest-1

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#3053]
_This issue was originally opened by @yydoow as hashicorp/terraform#17136. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hello,

I am trying to ran terraform upon the new launched China region cn-northwest-1 with following version
### Terraform Version

```
$ terraform -v
Terraform v0.11.2
+ provider.aws v1.7.0
+ provider.null v1.0.0

```
I also checked the github code, it seems with version I am running has the region supported 
https://github.com/terraform-providers/terraform-provider-aws/commit/8499d2e3651bfc8c7019175bf2f94ee6ee7a276c#diff-59e28844253678de7bce92be14f06597

I have no idea the issue here, could you please help me !

Thanks

### Crash Output

```
$ terraform plan
var.environment
  The environment of the resource

  Enter a value: dev

Acquiring state lock. This may take a few moments...
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.null_data_source.configuration: Refreshing state...

Error: Error refreshing state: 1 error(s) occurred:

* provider.aws: Not a valid region: cn-northwest-1
```

**Comment by yydoow** [comment]
My aws cli works fine upon the region, it looks like the provider issue
```
$ aws --region cn-northwest-1 s3 ls s3://apitrail
                           PRE AWSLogs/
```

**Comment by kwerey** [comment]
I hit this also in:

```
$ terraform -version
Terraform v0.11.2
+ provider.aws v1.7.1
```

Kinda looks like it just needs adding to accepted params for the ValidateRegion() function in config.go

```
// ValidateRegion returns an error if the configured region is not a
// valid aws region and nil otherwise.
func (c *Config) ValidateRegion() error {
	var regions = []string{
		"ap-northeast-1",
		"ap-northeast-2",
		"ap-south-1",
		"ap-southeast-1",
		"ap-southeast-2",
		"ca-central-1",
		"cn-north-1",
		"eu-central-1",
		"eu-west-1",
		"eu-west-2",
		"eu-west-3",
		"sa-east-1",
		"us-east-1",
		"us-east-2",
		"us-gov-west-1",
		"us-west-1",
		"us-west-2",
	}

	for _, valid := range regions {
		if c.Region == valid {
			return nil
		}
	}
	return fmt.Errorf("Not a valid region: %s", c.Region)
}
```

https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/config.go#L495

I also notice there's an IsChinaCloud flag that's currently only set for cn-north-1. I wouldn't be surprised if this needs setting for cn-northwest-1 to construct China service endpoints correctly:

```
func (c *AWSClient) IsChinaCloud() bool {
	if c.region == "cn-north-1" {
		return true
	}
	return false
}
```

https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/config.go#L223 

I don't have a go dev environment, but if it's quick and easy to set one up, I'll try fixing it and put in a PR if it really is this simple ;)

**Comment by bflad** [comment]
Support for this new region has been merged into master and will be released with v1.8.0, hopefully later today.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.8.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#180-january-29-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

Important note: if you're planning on using `cn-northwest-1` S3 for backend state, this PR (or newer version of it) needs to be merged and released in Terraform core (which will be 0.11.3+): https://github.com/hashicorp/terraform/pull/17216
